src/lib/wallet/MultipleAddressWallet.js

```diff
@@ -0,0 +1,51 @@
+// @flow
+import HDKey from 'hdkey'
+import Wallet from 'ethereumjs-wallet'
+import logger from '../logger/pino-logger'
+
+
+const log = logger.child({ from: 'MultipleAddressWallet' })
+
+type WalletsCollection = {
+  [key: string]: Wallet // Associative array
+}
+class MultipleAddressWallet {
+  ready: Promise<Web3>
+  wallet: Wallet
+  wallets: WalletsCollection
+  mnemonic: string
+  addresses: Array<string>
+  numOfAccounts: number = 10
+
+  constructor(mnemonic: string, numOfAccounts: number) {
+    logger.debug('MultipleAddressWallet ', { mnemonic }, { numOfAccounts })
+    this.numOfAccounts = numOfAccounts
+    this.mnemonic = mnemonic
+    this.addresses = []
+    this.wallets = {}
+  }
+
+  async ready() {
+    await this.initAccounts()
+    this.wallet = this.wallets[0]
```

Comment:
```suggestion
    this.wallet = this.wallets[this.accounts[0]]
```

Reply:
it's not resolved, i'll upload a fix manully

Reply:
fixed to 
`    this.wallet = this.wallets[this.addresses[0]]
`